XSRF and XSS countermeasures documented (#1814)
kyma-project/console: README.md

npm run start:api
```

> ### Follow security countermeasures

Reviewer: ```suggestion
### Security countermeasures
```

> When developing new features in Console UI, please respect the following rules that help reducing security related threats.

Reviewer: ```suggestion
When developing new features in Console UI, adhere to the following rules. This will help you to mitigate any security-related threats.
```

> #### Prevent XSRF (Cross-site request forgery)

Reviewer: ```suggestion
#### Prevent Cross-site request forgery (XSRF)
```
I would put the full name first and the abbreviation in the brackets

> - Do not store authentication token as a cookie. Make sure token is sent to the console backend service as a bearer token.

Reviewer: ```suggestion
- Do not store the authentication token as a cookie. Make sure the token is sent to the Console Backend Service as a bearer token.
```

> - Make sure that state-changing operations (gql mutations) are only trigered upon expolicit user interactions (e.g. form submissions).

Reviewer: ```suggestion
- Make sure that state-changing operations (gql mutations) are only triggered upon explicit user interactions such as form submissions.
```

> - Keep in mind that UI rendering (in response to user navigating between views) is only allowed to trigger read-only operations (gql querries and subscriptions) without any data mutations.

Reviewer: ```suggestion
- Keep in mind that UI rendering in response to user navigating between views is only allowed to trigger read-only operations (gql queries and subscriptions) without any data mutations.
```

> #### Protect against XSS (Cross-site scripting)

Reviewer: ```suggestion
#### Protect against Cross-site scripting (XSS)
```

> - It is recommended to use JS frameworks that have XSS prevention mechanisms built in (E.g. [reactJS](https://reactjs.org/docs/introducing-jsx.html#jsx-prevents-injection-attacks), [vue.js](https://vuejs.org/v2/guide/security.html#What-Vue-Does-to-Protect-You) or [angular](https://angular.io/guide/security#angulars-cross-site-scripting-security-model))

Reviewer: ```suggestion
- It is recommended to use JS frameworks that have built-in XSS prevention mechanisms, such as [reactJS](https://reactjs.org/docs/introducing-jsx.html#jsx-prevents-injection-attacks), [vue.js](https://vuejs.org/v2/guide/security.html#What-Vue-Does-to-Protect-You) or [angular](https://angular.io/guide/security#angulars-cross-site-scripting-security-model)).
```

> - As a rule of thumb, user-provided input can never be assumed 100% safe. Get familiar with prevention mechanismsm included in the framework of your choice. Make sure user input is sanitised before it is embedded in the DOM tree.

Reviewer: ```suggestion
- As a rule of thumb, you cannot perceive user input to be 100% safe. Get familiar with prevention mechanisms included in the framework of your choice. Make sure the user input is sanitized before it is embedded in the DOM tree.
```

> - Get familiar with most common [XSS bypasses and potential dangers](https://stackoverflow.com/questions/33644499/what-does-it-mean-when-they-say-react-is-xss-protected). Pay special attention to those when writing or reviewing the code.

Reviewer: ```suggestion
- Get familiar with the most common [XSS bypasses and potential dangers](https://stackoverflow.com/questions/33644499/what-does-it-mean-when-they-say-react-is-xss-protected). Keep them in mind when writing or reviewing the code.
```

> - Enable `Content-security-policy` header for all new microforntends to assure in-depth XSS prevention. Do not allow for `unsafe-eval` policy.

Reviewer: ```suggestion
- Enable the `Content-security-policy` header for all new micro frontends to ensure in-depth XSS prevention. Do not allow for `unsafe-eval` policy.
```
